dali/pipeline/operators/reader/loader/video_loader.h

```diff
@@ -182,12 +187,30 @@ class VideoLoader : public Loader<GPUBackend, SequenceWrapper> {
     int total_count = 1 + (count_ - 1) * stride_;
 
     for (size_t i = 0; i < file_label_pair_.size(); ++i) {
-      int frame_count = get_or_open_file(file_label_pair_[i].first).frame_count_;
+      const auto& file = get_or_open_file(file_label_pair_[i].first);
+      const auto stream = file.fmt_ctx_->streams[file.vid_stream_idx_];
+      int frame_count = file.frame_count_;
+
       for (int s = 0; s < frame_count && s + total_count <= frame_count; s += step_) {
-        frame_starts_.emplace_back(sequence_meta{i, s, file_label_pair_[i].second});
+        frame_starts_.emplace_back(sequence_meta{i, s, file_label_pair_[i].second,
+                                   codecpar(stream)->height, codecpar(stream)->width});
```

Comment:
I don't know how heavy is `codecpar(stream)` call, maybe get_or_open_file could return the video dimensions?

Reply:
It's not much
https://github.com/NVIDIA/DALI/blob/67997f5fb10b04d4581a66bf5e8f66d4bd3ee24f/dali/pipeline/operators/reader/loader/video_loader.cc#L36-L44